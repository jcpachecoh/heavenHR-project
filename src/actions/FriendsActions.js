import * as types from '../constants/ActionTypes';

export function addFriend(name, sex) {
  console.log("dentro funcion"+sex);
  return {
    type: types.ADD_FRIEND,
    name, 
    sex
  };
}

export function deleteFriend(id) {
  console.log(id);
  return {
    type: types.DELETE_FRIEND,
    id
  };
}

export function starFriend(id) {
  return {
    type: types.STAR_FRIEND,
    id
  };
}
