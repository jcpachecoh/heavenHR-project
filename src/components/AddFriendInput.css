:local(.addFriendInput) {
  border-radius: 0;
  border-color: #ABAAAA;
  border-left: 0;
  border-right: 0;
}

:local(.sexFriendInput) {
  
    padding: 8px;
    display: inline-block;
    width: 25%;
    }
  
    :local(.friendSex) {
      color: #999999;
      font-size: 16px;
      background: #fff;
      border-bottom: 1px solid #ccc;
    }

    :local(.messageStatus) {
      background: #FFC5C5
    }
    