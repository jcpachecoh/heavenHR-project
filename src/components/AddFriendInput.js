import React, { Component, PropTypes } from 'react';
import classnames from 'classnames';
import styles from './AddFriendInput.css';
import { messages } from '../constants/Messages'

class AddFriendInput extends Component {

  render () {
    return (<div>
      <input
        type="text"
        autoFocus="true"
        className={classnames('form-control', styles.addFriendInput)}
        placeholder="Type the name of a friend and press enter"
        value={this.state.name}
        onChange={this.handleChange.bind(this)}
        onKeyDown={this.handleSubmit.bind(this)} />
      <div className={classnames(styles.friendSex)}>

        <input 
        type="radio" 
        name="sex" 
        value="male" 
        className={classnames(styles.sexFriendInput)} 
        onChange={this.handleChangeSex.bind(this)}/>
        <img src="https://cdn4.iconfinder.com/data/icons/essential-part-3/32/216-Male-24.png" alt="male"/>

        <input type="radio" 
        name="sex"
        value="female" 
        className={classnames(styles.sexFriendInput)} 
        onChange={this.handleChangeSex.bind(this)}/>
        <img src="https://cdn4.iconfinder.com/data/icons/essential-part-3/32/217-Female-24.png" alt="female"/>
      </div>
      <div  className={classnames(styles.messageStatus)} >
        {this.state.status} 
      </div>
    </div>
    );
  }

  constructor (props, context) {
    super(props, context);
    this.state = {
      name: this.props.name || '',
      sex: this.props.sex,
      state: ""
    };
  }

  handleChange (e) {
    this.setState({ name: e.target.value, status: '' });
  }

  handleChangeSex(e) {
    this.setState({ sex: e.target.value, status: '' });
  }

  handleSubmit (e) {
    const name = this.state.name;
    const sex = this.state.sex;
   
    if(name && sex) {
      if (e.which === 13) {
        this.props.addFriend(name, sex);
        this.setState({ name: '', sex: '', status: messages.success });
      }
    }else {
      if(!name)
        this.setState({ status: messages.NameMessage });
      
      if(!sex)
        this.setState({ status: messages.sexMessage });
    }
  }

}

AddFriendInput.propTypes = {
  addFriend: PropTypes.func.isRequired
};

export default AddFriendInput
