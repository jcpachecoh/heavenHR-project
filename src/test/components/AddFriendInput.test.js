import React from 'react';
import AddFriendInput from '../../components/AddFriendInput';
import * as TestUtils from 'react-addons-test-utils';
import {addFriend} from '../../actions/FriendsActions';



describe('Testing <AddFriendInput /> component', () => {
    it('Should render AddFriendInput component ', () => {
        expect(TestUtils.createRenderer().render(
            <AddFriendInput addFriend={addFriend} />
          )).toMatchSnapshot()
    });

   /* describe('handle state when user enter a name', () => {
        it('should set state name when user type name', () => {
            const onKeyDown = sinon.spy();
            const wrapper = mount(<AddFriendInput addFriend={addFriend}/>);
            const input = wrapper.find('input');
            input.simulate('keyDown', {keyCode: 40});
            expect(onKeyDown.called).to.be.true;
        });
    });*/
});


