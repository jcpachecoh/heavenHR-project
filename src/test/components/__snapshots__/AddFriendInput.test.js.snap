// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <AddFriendInput /> component Should render AddFriendInput component  1`] = `
<div>
  <input
    autoFocus="true"
    className="form-control"
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="Type the name of a friend and press enter"
    type="text"
    value=""
  />
  <div
    className=""
  >
    <input
      className=""
      name="sex"
      onChange={[Function]}
      type="radio"
      value="male"
    />
    <img
      alt="male"
      src="https://cdn4.iconfinder.com/data/icons/essential-part-3/32/216-Male-24.png"
    />
    <input
      className=""
      name="sex"
      onChange={[Function]}
      type="radio"
      value="female"
    />
    <img
      alt="female"
      src="https://cdn4.iconfinder.com/data/icons/essential-part-3/32/217-Female-24.png"
    />
  </div>
  <div
    className=""
  />
</div>
`;
